src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/services/TestCaseDisplayService.kt

```diff
@@ -227,6 +234,25 @@ class TestCaseDisplayService(private val project: Project) {
 
         // insert test case components into selected class
         appendTestsToClass(testCaseComponents, selectedClass)
+
+        // schedule telemetry
+        val telemetryService = ApplicationManager.getApplication().getService(TestGenieTelemetryService::class.java)
```

Comment:
wouldn't it make sense to check for "opt-in" here, before mapping?

Reply:
I decided to keep all such checks in the telemetry service so that we never accidentally forget to do a check before calling the service. Mapping doesn't add much overhead at all, so I think it's fine.